website/pages/_app.js

```diff
@@ -59,11 +60,25 @@ export default function App({ Component, pageProps }) {
       {ALERT_BANNER_ACTIVE && (
         <AlertBanner {...alertBannerData} product="vault" hideOnMobile />
       )}
-      <HashiStackMenu />
-      <ProductSubnav />
-      <Component {...pageProps} />
-      <Footer openConsentManager={openConsentManager} />
+      <Layout {...(layoutData && { data: layoutData })}>
+        <Component {...pageProps} />
+      </Layout>
       <ConsentManager className="g-consent-manager" />
     </ErrorBoundary>
   )
 }
+
+App.getInitialProps = async ({ Component, ctx }) => {
+  const layoutQuery = Component.layout
+    ? Component.layout?.rivetParams ?? null
```

Comment:
They're not perfect, and it's not a blocker by any means, but here's what I put in the global types for `RivetParams`

```ts
interface RivetParams {
  query: string
  dependencies?: any[]
  variables?: Record<string, any>
  fragments?: any[]
}
```

Reply:
Will keep in mind, currently app is not a TS file.